app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -446,6 +450,25 @@ public void loadData(String baseURL, String data, String mimeType, String encodi
             geckoSession.loadData(data.getBytes(Charsets.UTF_8), mimeType, baseURL);
         }
 
+        @Override
+        public void getTelemetrySnapshots() {
+            final GeckoResponse<GeckoBundle> response = new GeckoResponse<GeckoBundle>() {
+                @Override
+                public void respond(GeckoBundle value) {
+                    if (value != null) {
+                        try {
+                            final JSONObject jsonData = value.toJSONObject();
+                            TelemetryWrapper.addMobileMetricsPing(jsonData);
+                        } catch (JSONException e) {
+                            e.printStackTrace();
```

Comment:
nit: Let's either log explicitly using "Log..." or if we think this should never happen then let's wrap the exception in an AssertionError and rethrow.